src/include/duckdb/storage/table/segment_tree.hpp

```diff
@@ -108,8 +120,24 @@ class SegmentTree {
 			return nodes[UnsafeNumericCast<idx_t>(index)].node.get();
 		}
 	}
+
+	const T *GetSegmentByIndex(int64_t index) const {
+		auto l = Lock();
+		return GetSegmentByIndex(l, index);
+	}
+	const T *GetSegmentByIndex(SegmentLock &l, int64_t index) const {
+		auto &non_const_this = (SegmentTree &)(*this); // NOLINT
```

Comment:
Let's remove these const casts and just duplicate the methods

Reply:
Done 👍  <https://github.com/duckdb/duckdb/pull/15406/commits/13c7fad86956beef5c25e72f497fba8b84ccd3b8>

Please have a look at the commit, I'm still on the fence whether this is better or worse than solving it with a const cast

Reply:
`const_cast` is undefined behavior - so not using it is always better :)

Reply:
I've cleaned up some of the duplicate code in https://github.com/duckdb/duckdb/pull/15406/commits/64e4832463443373cceae43086ce6bcfdff5ba5c but I think we do need some of it unfortunately